src/dioptra/restapi/v1/schemas.py

```diff
@@ -24,63 +24,110 @@
+    schema = {
+        "id": fields.Integer(
+            attribute="id",
+            metadata=dict(description=f"ID for the {name} resource."),
+            dump_only=True,
+        ),
+        "snapshotId": fields.Integer(
+            attribute="snapshot_id",
+            metadata=dict(description=f"ID for the underlying {name} snapshot."),
+            dump_only=True,
+        ),
+        "groupId": fields.Integer(
+            attribute="group_id",
+            metadata=dict(
+                description=f"ID of the Group that will own the {name} resource."
             ),
-            "createdOn": fields.DateTime(
-                attribute="created_on",
-                metadata=dict(
-                    description=f"Timestamp when the {name} resource was created."
-                ),
-                dump_only=True,
+            load_only=True,
+        ),
+        "group": fields.Nested(
+            GroupRefSchema,
+            attribute="group",
+            metadata=dict(description=f"Group that owns the {name} resource."),
+            dump_only=True,
+        ),
+        "user": fields.Nested(
+            UserRefSchema,
+            attribute="user",
+            metadata=dict(description=f"User that created the {name} resource."),
+            dump_only=True,
+        ),
+        "createdOn": fields.DateTime(
+            attribute="created_on",
+            metadata=dict(
+                description=f"Timestamp when the {name} resource was created."
             ),
-            "lastModifiedOn": fields.DateTime(
-                attribute="last_modified_on",
-                metadata=dict(
-                    description=f"Timestamp when the {name} resource was last modified."
-                ),
-                dump_only=True,
+            dump_only=True,
+        ),
+        "lastModifiedOn": fields.DateTime(
+            attribute="last_modified_on",
+            metadata=dict(
+                description=f"Timestamp when the {name} resource was last modified."
             ),
-            "tags": fields.Nested(
-                TagRefSchema,
-                attribute="tags",
-                metadata=dict(description="Tags associated with the {name} resource."),
-                many=True,
-                dump_only=True,
+            dump_only=True,
+        ),
+        "latestSnapshot": fields.Bool(
+            attribute="latest_snapshot",
+            metadata=dict(
+                description=f"Whether or not the {name} resource is the latest version."
             ),
-        }
-    )
+            dump_only=True,
+        ),
+        "tags": fields.Nested(
+            TagRefSchema,
+            attribute="tags",
+            metadata=dict(description="Tags associated with the {name} resource."),
+            many=True,
+            dump_only=True,
+        ),
+    }
+
+    if not snapshot:
+        schema.pop("snapshotId")
+        schema.pop("user")
+        schema.pop("createdOn")
```

Comment:
We should keep the `createdOn` field for regular resources.

Reply:
fixed